pkg/controller/toolchainstatus/toolchainstatus_controller.go

```diff
@@ -0,0 +1,438 @@
+	if err != nil {
+		err = errs.Wrap(err, status.ErrMsgCannotGetDeployment)
+		errCondition := status.NewComponentErrorCondition(toolchainv1alpha1.ToolchainStatusDeploymentNotFoundReason, err.Error())
+		operatorStatus.Conditions = []toolchainv1alpha1.Condition{*errCondition}
+		toolchainStatus.Status.HostOperator = operatorStatus
+		return err
+	}
+	operatorStatus.DeploymentName = hostOperatorDeploymentName
+
+	// check host operator deployment status
+	deploymentConditions, err := status.GetDeploymentStatusConditions(r.client, hostOperatorDeploymentName, toolchainStatus.Namespace)
+
+	// update toolchainStatus
+	operatorStatus.Conditions = deploymentConditions
+	toolchainStatus.Status.HostOperator = operatorStatus
+	return err
+}
+
+// registrationServiceHandleStatus retrieves the Deployment for the registration service and adds its status to ToolchainStatus. It returns an error
+// if the registration service is not ready
+func (r *ReconcileToolchainStatus) registrationServiceHandleStatus(reqLogger logr.Logger, toolchainStatus *toolchainv1alpha1.ToolchainStatus) (err error) {
+
+	s := regServiceSubstatusHandler{
+		controllerClient: r.client,
+		httpClientImpl:   r.httpClientImpl,
+	}
+
+	// gather the functions for handling registration service status eg. resource templates, deployment, health endpoint
+	substatusHandlers := []statusHandlerFunc{
+		s.addRegistrationServiceResourceStatus,
+		s.addRegistrationServiceDeploymentStatus,
+		s.addRegistrationServiceHealthStatus,
+	}
+
+	// call each of the registration service status handlers
+	for _, handler := range substatusHandlers {
+		handlerErr := handler(reqLogger, toolchainStatus)
+		if handlerErr != nil {
+			err = handlerErr
+		}
+	}
+
+	return err
+}
+
+// memberHandleStatus retrieves the status of member clusters and adds them to ToolchainStatus. It returns an error
+// if any of the members are not ready or if no member clusters are found
+func (r *ReconcileToolchainStatus) membersHandleStatus(reqLogger logr.Logger, toolchainStatus *toolchainv1alpha1.ToolchainStatus) (componentError error) {
+	// get member clusters
+	memberClusters := r.getMembersFunc()
+	members := []toolchainv1alpha1.Member{}
+	if len(memberClusters) == 0 {
+		err := fmt.Errorf("no member clusters found")
+		memberStatusNotFoundCondition := status.NewComponentErrorCondition(toolchainv1alpha1.ToolchainStatusMemberStatusNoClustersFoundReason, err.Error())
+		memberStatus := customMemberStatus(*memberStatusNotFoundCondition)
+		noMemberClusterStatus := toolchainv1alpha1.Member{
+			ClusterName:  "",
+			MemberStatus: memberStatus,
+		}
+		members = append(members, noMemberClusterStatus)
+		return err
+	}
+	for _, cluster := range memberClusters {
+		memberClient := cluster.Client
+		if memberClient == nil {
+			// there is no client for the member cluster, create a status condition and add it to this member's status
+			err := fmt.Errorf("member cluster does not have a client assigned")
+			componentError = err
```

Comment:
```suggestion
			componentError = fmt.Errorf("member cluster does not have a client assigned")
```

Reply:
Nice catch, done in 4c61f7dc04914b4876dda060b2ae2c4460567717